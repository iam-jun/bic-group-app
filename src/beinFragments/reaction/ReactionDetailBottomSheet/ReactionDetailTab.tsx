import React, { FC, useEffect, useState } from 'react';
import {
  View, StyleSheet, Dimensions, FlatList,
} from 'react-native';

import { ReactionType } from '~/constants/reactions';
import PrimaryItem from '~/beinComponents/list/items/PrimaryItem';
import LoadingIndicator from '~/beinComponents/LoadingIndicator';
import spacing from '~/theme/spacing';
import { IReaction } from '~/interfaces/IPost';

export interface ReactionDetailTabProps {
  reactionType: ReactionType;
  limit?: number;
  height?: number;
  onPressItem?: (item: any) => void;

  getDataPromise?: any;
  getDataParam?: any;
}

const screenHeight = Dimensions.get('window').height;
const contentBarHeight = 0.6 * screenHeight;

const ReactionDetailTab: FC<ReactionDetailTabProps> = ({
  reactionType,
  limit = 1,
  height = contentBarHeight,
  onPressItem,
  getDataPromise,
  getDataParam,
}: ReactionDetailTabProps) => {
  const [data, setData] = useState<IReaction[]>([]);
  const [loading, setLoading] = useState(false);

  const getData = () => {
    if (getDataPromise && getDataParam) {
      setLoading(true);
      const param = { ...getDataParam, reactionName: reactionType, limit };
      getDataPromise?.(param)
        ?.then?.((data: any) => {
          setData(data || []);
          setLoading(false);
        })
        .catch((e: any) => {
          console.log(`\x1b[31m🐣️ ReactionDetailTab get error ${e}\x1b[0m`);
          setLoading(false);
        });
    }
  };

  useEffect(
    () => {
      setData([]);
      getData();
    }, [reactionType],
  );

  const _onPressItem = (item: any) => {
    onPressItem?.(item);
  };

  const onLoadMore = () => {
    if (getDataPromise && getDataParam && !!data?.[0]) {
      const param = {
        ...getDataParam, reactionName: reactionType, limit, latestId: data[data.length - 1].reactionId, order: 'ASC',
      };
      getDataPromise?.(param)
        ?.then?.((_data: any) => {
          setData(data.concat(_data || []));
        })
        .catch((e: any) => {
          console.log(`\x1b[31m🐣️ ReactionDetailTab get more error ${e}\x1b[0m`);
        });
    }
  }

  const renderItem = (item: any) => (
    <PrimaryItem
      testID={`reaction_detail_bottomSheet.${item?.item?.fullname}`}
      showAvatar
      height={44}
      onPress={() => _onPressItem(item)}
      avatar={item?.item?.avatar}
      title={item?.item?.fullname}
    />
  );

  const renderFooter = () => {
    if (loading) {
      return <LoadingIndicator />;
    }
    return null;
  };

  const renderHeader = () => <View style={styles.header} />;

  return (
    <View style={{ height: 100 }}>
      <FlatList
        testID="reaction_detail_bottomSheet.list_user"
        style={styles.listContainer}
        data={data}
        renderItem={renderItem}
        ListHeaderComponent={renderHeader}
        ListFooterComponent={renderFooter}
        onEndReached={onLoadMore}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  listContainer: {
    paddingTop: spacing.padding.tiny,
  },
  header: {
    paddingTop: spacing.padding.tiny,
  },
});

export default ReactionDetailTab;
