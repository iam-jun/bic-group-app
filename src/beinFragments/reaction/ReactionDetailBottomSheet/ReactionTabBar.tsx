import React, {FC, useState, useEffect, useRef} from 'react';
import {FlatList, View, StyleSheet, Platform} from 'react-native';
import {useTheme} from 'react-native-paper';

import {ITheme} from '~/theme/interfaces';
import {blacklistReactions, ReactionType} from '~/constants/reactions';

import Text from '~/beinComponents/Text';
import Button from '~/beinComponents/Button';
import NodeEmoji from 'node-emoji';

export interface ReactionTabBarProps {
  initReaction?: ReactionType;
  reactionCounts: any;
  onChangeTab?: (item: any, index: number) => void;
}

const itemWidth = 80;

const ReactionTabBar: FC<ReactionTabBarProps> = ({
  reactionCounts,
  initReaction,
  onChangeTab,
}: ReactionTabBarProps) => {
  const [data, setData] = useState([]);
  const [activeIndex, setActiveIndex] = useState(-1);
  const flatListRef = useRef<any>();

  const theme = useTheme() as ITheme;
  const {colors} = theme;
  const styles = createStyle(theme);

  useEffect(() => {
    onChangeTab?.(data?.[activeIndex], activeIndex);
  }, [activeIndex]);

  const _onPressTab = (index: number) => {
    setActiveIndex(index);
    if (data.length > index) {
      flatListRef?.current?.scrollToIndex?.({
        index: index,
        animated: true,
        viewPosition: 0.5,
      });
    }
  };

  useEffect(() => {
    //wait for set data finish, avoid scroll wrong position of tab
    setTimeout(() => {
      if (data?.length > 0 && initReaction) {
        data.map((item: any, index) => {
          if (item?.reactionType === initReaction) {
            _onPressTab(index);
          }
        });
      }
    }, 300);
  }, [data, initReaction]);

  useEffect(() => {
    if (reactionCounts) {
      const reaactionCountMap = new Map();
      const newData: any = [];
      Object.values(reactionCounts || {})?.map((reaction: any) => {
        const key = Object.keys(reaction || {})?.[0];
        if (key) {
          reaactionCountMap.set(key, reaction?.[key]);
        }
      });

      for (const [key, value] of reaactionCountMap) {
        const reactionType = key as ReactionType;
        if (!blacklistReactions?.[reactionType] && value > 0) {
          newData.push({
            reactionType,
            count: value,
          });
        }
      }
      setData(newData);
    } else {
      //reset
      setData([]);
      setActiveIndex(-1);
    }
  }, [reactionCounts]);

  const onScrollToIndexFailed = () => {
    console.log(`\x1b[31m🐣️ ReactionTabBar onScrollToIndexFailed\x1b[0m`);
  };

  const renderItem = ({item, index}: any) => {
    const {reactionType, count} = item || {};
    const isActive = activeIndex === index;
    const emoji = NodeEmoji.find(reactionType)?.emoji || '';
    return (
      <View>
        <Button
          testID={`reaction_detail_bottomSheet.${reactionType}`}
          style={styles.tabItem}
          onPress={() => _onPressTab(index)}>
          <Text.H5 color={isActive ? colors.primary7 : colors.textPrimary}>
            {emoji}
          </Text.H5>
          <Text.H6 color={isActive ? colors.primary7 : colors.textPrimary}>
            {` ${count}`}
          </Text.H6>
        </Button>
        {isActive && (
          <View
            testID={`reaction_detail_bottomSheet.active_${reactionType}`}
            style={styles.tabItemActive}
          />
        )}
      </View>
    );
  };

  return (
    <FlatList
      ref={flatListRef}
      horizontal
      data={data}
      snapToAlignment={'center'}
      showsHorizontalScrollIndicator={false}
      getItemLayout={(data, index) => ({
        length: itemWidth,
        offset: itemWidth * index,
        index,
      })}
      onScrollToIndexFailed={onScrollToIndexFailed}
      snapToInterval={itemWidth}
      style={styles.container}
      renderItem={renderItem}
      keyExtractor={(item, index) => `reaction_tab_${index}`}
    />
  );
};

const createStyle = (theme: ITheme) => {
  const {colors, spacing} = theme;
  return StyleSheet.create({
    container: {
      borderBottomWidth: 1,
      borderColor: colors.borderDisable,
      maxWidth: 500,
    },
    tabItem: {
      width: itemWidth,
      height: 44,
      justifyContent: 'center',
      alignItems: 'center',
      flexDirection: 'row',
    },
    tabItemActive: {
      position: 'absolute',
      bottom: 0,
      width: itemWidth,
      height: 2,
      backgroundColor: colors.primary7,
    },
  });
};

export default ReactionTabBar;
