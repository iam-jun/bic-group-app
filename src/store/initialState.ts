import {appInitState} from '~/store/app/reducer';
import {modalInitState} from '~/store/modal/reducer';
import {authInitState} from '~/screens/Auth/redux/reducer';
import {postInitState} from '~/screens/Post/redux/reducer';
import {groupInitState} from '~/screens/Groups/redux/reducer';
import {homeInitState} from '~/screens/Home/redux/reducer';
import {notiInitState} from '~/screens/Notification/redux/reducer';
import {menuInitState} from '~/screens/Menu/redux/reducer';
import {noInternetInitState} from '~/screens/NoInternet/redux/reducer';
import {mentionInputInitState} from '~/beinComponents/inputs/MentionInput/redux/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
};

export default initialState;
