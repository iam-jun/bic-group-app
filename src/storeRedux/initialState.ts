import { appInitState } from '~/storeRedux/app/reducer';
import { modalInitState } from '~/storeRedux/modal/reducer';
import { authInitState } from '~/storeRedux/auth/reducer';
import { postInitState } from '~/storeRedux/post/reducer';
import { groupInitState } from '~/storeRedux/groups/reducer';
import { homeInitState } from '~/storeRedux/home/reducer';
import { notiInitState } from '~/storeRedux/notification/reducer';
import { menuInitState } from '~/storeRedux/menu/reducer';
import { noInternetInitState } from '~/storeRedux/network/reducer';
import { mentionInputInitState } from '~/beinComponents/inputs/MentionInput/redux/reducer';
import { giphyInitState } from './giphy/reducer';

const initialState = {
  app: appInitState,
  modal: modalInitState,
  auth: authInitState,
  post: postInitState,
  groups: groupInitState,
  home: homeInitState,
  notifications: notiInitState,
  menu: menuInitState,
  noInternet: noInternetInitState,
  mentionInput: mentionInputInitState,
  giphy: giphyInitState,
};

export default initialState;
