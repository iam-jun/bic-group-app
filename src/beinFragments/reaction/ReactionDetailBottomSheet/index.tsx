import React, {useRef, useState, useEffect} from 'react';
import {Dimensions, View} from 'react-native';

import BottomSheet from '~/beinComponents/BottomSheet';
import {useKeySelector} from '~/hooks/selector';
import {useDispatch} from 'react-redux';
import modalKeySelector from '~/store/modal/selectors';
import {clearReactionDetailBottomSheet} from '~/store/modal/actions';
import ReactionTabBar from '~/beinFragments/reaction/ReactionDetailBottomSheet/ReactionTabBar';
import ReactionDetailTab from '~/beinFragments/reaction/ReactionDetailBottomSheet/ReactionDetailTab';
import menuActions from '~/screens/Menu/redux/actions';
import {useUserIdAuth} from '~/hooks/auth';
import {useRootNavigation} from '~/hooks/navigation';
import homeStack from '~/router/navigator/MainStack/HomeStack/stack';

const screenHeight = Dimensions.get('window').height;
const contentBarHeight = 0.6 * screenHeight;

const ReactionDetailBottomSheet = () => {
  const reactionSheetRef: any = useRef();
  const [selectingReaction, setSelectingReaction] = useState<any>();

  const dispatch = useDispatch();
  const {rootNavigation} = useRootNavigation();

  const userId = useUserIdAuth();
  const data = useKeySelector(modalKeySelector.reactionDetailBottomSheet);
  const {isOpen, reactionCounts, postId, commentId, initReaction} = data || {};

  useEffect(() => {
    //reset
    if (!reactionCounts) {
      setSelectingReaction(undefined);
    }
  }, [reactionCounts]);

  const _onClose = () => {
    dispatch(clearReactionDetailBottomSheet());
  };

  const onChangeTab = (item: any) => {
    if (item?.reactionType) {
      setSelectingReaction(item?.reactionType);
    }
  };

  const onPressItem = (item: any) => {
    const itemUserId = item?.item?.user?.id;
    if (itemUserId) {
      dispatch(
        menuActions.selectedProfile({
          id: itemUserId,
          isPublic: userId !== itemUserId,
        }),
      );
      rootNavigation.navigate(homeStack.publicProfile);
    }
    reactionSheetRef?.current?.close?.();
  };

  return (
    <BottomSheet
      modalizeRef={reactionSheetRef}
      isOpen={isOpen}
      side="center"
      menuMinWidth={375}
      onClose={_onClose}
      ContentComponent={
        <View>
          <ReactionTabBar
            reactionCounts={reactionCounts}
            onChangeTab={onChangeTab}
            initReaction={initReaction}
          />
          <ReactionDetailTab
            reactionType={selectingReaction}
            postId={postId}
            height={contentBarHeight}
            onPressItem={onPressItem}
          />
        </View>
      }
    />
  );
};

export default ReactionDetailBottomSheet;
