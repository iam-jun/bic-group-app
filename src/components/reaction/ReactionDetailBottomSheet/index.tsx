import React, { useRef, useState, useEffect } from 'react';
import { Dimensions, View } from 'react-native';

import { useDispatch } from 'react-redux';
import BottomSheet from '~/beinComponents/BottomSheet';
import { useKeySelector } from '~/hooks/selector';
import { clearReactionDetailBottomSheet } from '~/storeRedux/modal/actions';
import ReactionTabBar from '~/components/reaction/ReactionDetailBottomSheet/ReactionTabBar';
import ReactionDetailTab from '~/components/reaction/ReactionDetailBottomSheet/ReactionDetailTab';
import { useRootNavigation } from '~/hooks/navigation';
import mainStack from '~/router/navigator/MainStack/stack';
import modalKeySelector from '~/storeRedux/modal/keySelector';

const screenHeight = Dimensions.get('window').height;
const contentBarHeight = 0.6 * screenHeight;

const ReactionDetailBottomSheet = () => {
  const reactionSheetRef: any = useRef();
  const [selectingReaction, setSelectingReaction] = useState<any>();

  const dispatch = useDispatch();
  const { rootNavigation } = useRootNavigation();

  const data = useKeySelector(modalKeySelector.reactionDetailBottomSheet);
  const {
    isOpen, reactionCounts, initReaction, getDataPromise, getDataParam,
  } = data || {};

  useEffect(
    () => {
    // reset
      if (!reactionCounts) {
        setSelectingReaction(undefined);
      }
    }, [reactionCounts],
  );

  const _onClose = () => {
    dispatch(clearReactionDetailBottomSheet());
  };

  const onChangeTab = (item: any) => {
    if (item?.reactionType) {
      setSelectingReaction(item?.reactionType);
    }
  };

  const onPressItem = (item: any) => {
    const itemUserId = item?.item?.id;
    if (itemUserId) {
      rootNavigation.navigate(
        mainStack.userProfile, { userId: itemUserId },
      );
    } else {
      rootNavigation.navigate(
        mainStack.userProfile, {
          userId: item?.item.username,
          params: {
            type: 'username',
          },
        },
      );
    }
    reactionSheetRef?.current?.close?.();
  };

  return (
    <BottomSheet
      modalizeRef={reactionSheetRef}
      isOpen={isOpen}
      onClose={_onClose}
      ContentComponent={(
        <View>
          <ReactionTabBar
            reactionCounts={reactionCounts}
            onChangeTab={onChangeTab}
            initReaction={initReaction}
          />
          <ReactionDetailTab
            reactionType={selectingReaction || initReaction}
            height={contentBarHeight}
            onPressItem={onPressItem}
            getDataPromise={getDataPromise}
            getDataParam={getDataParam}
          />
        </View>
      )}
    />
  );
};

export default ReactionDetailBottomSheet;
