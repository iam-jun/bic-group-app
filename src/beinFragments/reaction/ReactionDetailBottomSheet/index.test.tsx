import * as React from 'react';
import {cleanup, waitFor, act} from '@testing-library/react-native';

import {
  fireEvent,
  renderWithRedux,
  configureStore,
  waitForComponentToPaint,
} from '~/test/testUtils';
import ReactionDetailBottomSheet from './index';
import initialState from '~/store/initialState';
// import {waitFor} from 'react-test-renderer';

afterEach(cleanup);

describe('ReactionDetailBottomSheet component', () => {
  const mockStore = configureStore([]);

  const listUserReact = [
    {
      avatar:
        'https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/75f92c17-843b-49de-8445-93531854fd65.jpeg',
      fullname: 'Hoàng Minh Trọng',
      id: '53',
    },
    {
      avatar:
        'https://bein-entity-attribute-stg.s3.ap-southeast-1.amazonaws.com/user/avatar/Avatar_Profile.png',
      fullname: 'Nguyễn Thị Ngọc Linh',
      id: '33',
    },
    {
      avatar:
        'https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/363963d1-e89a-4899-855e-44b1f66401dc.jpg',
      fullname: 'Nguyen Thi Thu Queen',
      id: '58',
    },
  ];
  const getDataPromise = jest
    .fn()
    .mockImplementation((): Promise<any> => Promise.resolve(listUserReact));

  const fake_data = {
    getDataParam: {
      commentId: undefined,
      postId: 'efe9c800-92be-11ec-8080-800124087730',
    },
    getDataPromise: getDataPromise,
    initReaction: 'grinning_face_with_star_eyes',
    isOpen: true,
    reactionCounts: {
      comment: 9,
      comment_count: 11,
      grinning_face_with_star_eyes: 2,
      kissing_closed_eyes: 2,
      kissing_smiling_eyes: 2,
      sweat_smile: 3,
    },
  };
  it('renders correctly', async () => {
    const storeData = {...initialState};

    // @ts-ignore
    storeData.modal.reactionDetailBottomSheet = fake_data;
    const store = mockStore(storeData);
    const rendered = await waitFor(() =>
      renderWithRedux(<ReactionDetailBottomSheet />, store),
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it(`should render list user react correspondingly when clicking reaction item on ReactionTabBar`, async () => {
    const storeData = {...initialState};

    // @ts-ignore
    storeData.modal.reactionDetailBottomSheet = fake_data;
    const store = mockStore(storeData);

    const rendered = await waitFor(() =>
      renderWithRedux(<ReactionDetailBottomSheet />, store),
    );

    const btnComponent = rendered.getByTestId(
      'reaction_detail_bottomSheet.kissing_closed_eyes',
    );
    expect(btnComponent).toBeDefined();
    fireEvent.press(btnComponent);
  });

  it(`should render list empty user react if call api failed`, async () => {
    const getDataPromiseFail = jest
      .fn()
      .mockImplementation((): Promise<any> => Promise.reject());

    const storeData = {...initialState};

    // @ts-ignore
    storeData.modal.reactionDetailBottomSheet = {
      ...fake_data,
      getDataPromise: getDataPromiseFail,
      isOpen: false,
    };
    const store = mockStore(storeData);

    const rendered = await waitFor(() =>
      renderWithRedux(<ReactionDetailBottomSheet />, store),
    );
    const listUserComponent = rendered.getByTestId(
      'reaction_detail_bottomSheet.list_user',
    );

    expect(listUserComponent.props?.data?.length).toEqual(0);
  });

  it(`should call navigate to user profile by id when click item`, async () => {
    const storeData = {...initialState};

    // @ts-ignore
    storeData.modal.reactionDetailBottomSheet = fake_data;
    const store = mockStore(storeData);

    const rendered = await waitFor(() =>
      renderWithRedux(<ReactionDetailBottomSheet />, store),
    );

    const itemComponent = rendered.getByTestId(
      'reaction_detail_bottomSheet.Hoàng Minh Trọng',
    );
    expect(itemComponent).toBeDefined();
    fireEvent.press(itemComponent);
  });

  it(`should call navigate to user profile by username when click item`, async () => {
    const storeData = {...initialState};
    const _getDataPromise = jest.fn().mockImplementation(
      (): Promise<any> =>
        Promise.resolve([
          {
            avatar:
              'https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/75f92c17-843b-49de-8445-93531854fd65.jpeg',
            fullname: 'Hoàng Minh Trọng',
            username: 'tronghoang',
            id: '',
          },
        ]),
    );

    // @ts-ignore
    storeData.modal.reactionDetailBottomSheet = {
      ...fake_data,
      getDataPromise: _getDataPromise,
    };
    const store = mockStore(storeData);

    const rendered = await waitFor(() =>
      renderWithRedux(<ReactionDetailBottomSheet />, store),
    );

    const itemComponent = rendered.getByTestId(
      'reaction_detail_bottomSheet.Hoàng Minh Trọng',
    );
    expect(itemComponent).toBeDefined();
    fireEvent.press(itemComponent);
  });

  it(`should render list empty user react if call api return null`, async () => {
    const getDataPromiseFail = jest
      .fn()
      .mockImplementation((): Promise<any> => Promise.resolve(null));

    const storeData = {...initialState};

    // @ts-ignore
    storeData.modal.reactionDetailBottomSheet = {
      ...fake_data,
      getDataPromise: getDataPromiseFail,
      isOpen: false,
    };
    const store = mockStore(storeData);

    const rendered = await waitFor(() =>
      renderWithRedux(<ReactionDetailBottomSheet />, store),
    );
    const listUserComponent = rendered.getByTestId(
      'reaction_detail_bottomSheet.list_user',
    );

    expect(listUserComponent.props?.data?.length).toEqual(0);
  });
});
