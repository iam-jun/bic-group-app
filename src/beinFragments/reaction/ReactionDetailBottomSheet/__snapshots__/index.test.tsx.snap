// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionDetailBottomSheet component renders correctly 1`] = `
<View>
  <View>
    <View>
      <RCTScrollView
        data={Array []}
        getItem={[Function]}
        getItemCount={[Function]}
        getItemLayout={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onScrollToIndexFailed={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        snapToInterval={80}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#A4A9B7",
          }
        }
        viewabilityConfigCallbackPairs={Array []}
      >
        <View />
      </RCTScrollView>
      <View
        style={
          Object {
            "height": 800.4,
          }
        }
      >
        <RCTScrollView
          ListFooterComponent={[Function]}
          ListHeaderComponent={[Function]}
          data={
            Array [
              Object {
                "avatar": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/75f92c17-843b-49de-8445-93531854fd65.jpeg",
                "fullname": "Hoàng Minh Trọng",
                "id": "53",
              },
              Object {
                "avatar": "https://bein-entity-attribute-stg.s3.ap-southeast-1.amazonaws.com/user/avatar/Avatar_Profile.png",
                "fullname": "Nguyễn Thị Ngọc Linh",
                "id": "33",
              },
              Object {
                "avatar": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/363963d1-e89a-4899-855e-44b1f66401dc.jpg",
                "fullname": "Nguyen Thi Thu Queen",
                "id": "58",
              },
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onEndReached={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "paddingTop": 4,
            }
          }
          testID="reaction_detail_bottomSheet.list_user"
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
            >
              <View
                style={
                  Object {
                    "paddingTop": 4,
                  }
                }
              />
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 44,
                      "opacity": 1,
                      "paddingHorizontal": 12,
                    }
                  }
                  testID="reaction_detail_bottomSheet.Hoàng Minh Trọng"
                >
                  <View
                    style={
                      Object {
                        "borderRadius": 4,
                        "height": 40,
                        "marginRight": 12,
                        "width": 40,
                      }
                    }
                    testID="avatar"
                  >
                    <View
                      style={
                        Object {
                          "borderRadius": 4,
                          "height": 40,
                          "width": 40,
                        }
                      }
                      testID="avatar_container"
                    >
                      <View
                        onLayout={[Function]}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "position": "relative",
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#f9f9f9",
                                "borderRadius": 4,
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              }
                            }
                            testID="Image__placeholder"
                          />
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Object {
                                "borderRadius": 4,
                                "height": 40,
                                "width": 40,
                              },
                            ]
                          }
                        >
                          <FastImageView
                            onFastImageError={[Function]}
                            onFastImageLoadEnd={[Function]}
                            resizeMode="cover"
                            source={
                              Object {
                                "uri": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/75f92c17-843b-49de-8445-93531854fd65.jpeg",
                              }
                            }
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                            testID="avatar.image"
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "color": "#181C32",
                            "fontFamily": "BeVietnamPro-SemiBold",
                            "fontSize": 13,
                          },
                          Object {},
                          undefined,
                        ]
                      }
                    >
                      Hoàng Minh Trọng
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 44,
                      "opacity": 1,
                      "paddingHorizontal": 12,
                    }
                  }
                  testID="reaction_detail_bottomSheet.Nguyễn Thị Ngọc Linh"
                >
                  <View
                    style={
                      Object {
                        "borderRadius": 4,
                        "height": 40,
                        "marginRight": 12,
                        "width": 40,
                      }
                    }
                    testID="avatar"
                  >
                    <View
                      style={
                        Object {
                          "borderRadius": 4,
                          "height": 40,
                          "width": 40,
                        }
                      }
                      testID="avatar_container"
                    >
                      <View
                        onLayout={[Function]}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "position": "relative",
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#f9f9f9",
                                "borderRadius": 4,
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              }
                            }
                            testID="Image__placeholder"
                          />
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Object {
                                "borderRadius": 4,
                                "height": 40,
                                "width": 40,
                              },
                            ]
                          }
                        >
                          <FastImageView
                            onFastImageError={[Function]}
                            onFastImageLoadEnd={[Function]}
                            resizeMode="cover"
                            source={
                              Object {
                                "uri": "https://bein-entity-attribute-stg.s3.ap-southeast-1.amazonaws.com/user/avatar/Avatar_Profile.png",
                              }
                            }
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                            testID="avatar.image"
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "color": "#181C32",
                            "fontFamily": "BeVietnamPro-SemiBold",
                            "fontSize": 13,
                          },
                          Object {},
                          undefined,
                        ]
                      }
                    >
                      Nguyễn Thị Ngọc Linh
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View>
                <View
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 44,
                      "opacity": 1,
                      "paddingHorizontal": 12,
                    }
                  }
                  testID="reaction_detail_bottomSheet.Nguyen Thi Thu Queen"
                >
                  <View
                    style={
                      Object {
                        "borderRadius": 4,
                        "height": 40,
                        "marginRight": 12,
                        "width": 40,
                      }
                    }
                    testID="avatar"
                  >
                    <View
                      style={
                        Object {
                          "borderRadius": 4,
                          "height": 40,
                          "width": 40,
                        }
                      }
                      testID="avatar_container"
                    >
                      <View
                        onLayout={[Function]}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "position": "relative",
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#f9f9f9",
                                "borderRadius": 4,
                                "height": 40,
                                "justifyContent": "center",
                                "width": 40,
                              }
                            }
                            testID="Image__placeholder"
                          />
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Object {
                                "borderRadius": 4,
                                "height": 40,
                                "width": 40,
                              },
                            ]
                          }
                        >
                          <FastImageView
                            onFastImageError={[Function]}
                            onFastImageLoadEnd={[Function]}
                            resizeMode="cover"
                            source={
                              Object {
                                "uri": "https://bein-entity-attribute-sandbox.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/363963d1-e89a-4899-855e-44b1f66401dc.jpg",
                              }
                            }
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                            testID="avatar.image"
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "color": "#181C32",
                            "fontFamily": "BeVietnamPro-SemiBold",
                            "fontSize": 13,
                          },
                          Object {},
                          undefined,
                        ]
                      }
                    >
                      Nguyen Thi Thu Queen
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              onLayout={[Function]}
            />
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
